src/pkg/retention/policy/rule/nothing/evaluator.go

```diff
@@ -0,0 +1,42 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nothing
+
+import (
+	"github.com/goharbor/harbor/src/pkg/retention/policy/action"
+	"github.com/goharbor/harbor/src/pkg/retention/policy/rule"
+	"github.com/goharbor/harbor/src/pkg/retention/res"
+)
+
+const (
+	// TemplateID of the always retain rule
+	TemplateID = "nothing"
+)
+
+type evaluator struct{}
+
+// Process for the "always" Evaluator simply returns the input with no error
+func (e *evaluator) Process(artifacts []*res.Candidate) ([]*res.Candidate, error) {
+	return nil, nil
```

Comment:
not `nil`, should be an empty array

or use the declared variable in the return statement

```
Process(artifacts []*res.Candidate) (processed []*res.Candidate, err error) {
    return processed
}
```